task/oci-copy/0.1/oci-copy.yaml

```diff
@@ -71,16 +75,33 @@ spec:
         capabilities:
           add:
             - SETFCAP
+      env:
+      - name: BEARER_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: $(params.BEARER_TOKEN_SECRET_NAME)
+            key: token
+            optional: true
       script: |
-        set -eu
+        set -e
         set -o pipefail
 
+        CURL_ARGS=()
+        if [ -n "${BEARER_TOKEN}" ]; then
+          echo "Found bearer token. Using it for authentication."
+          CURL_ARGS+=(-H "Authorization: Bearer ${BEARER_TOKEN}")
+        else
+          echo "Proceeding with anonymous requests"
+        fi
+
+        set -u
+
         for varfile in /var/workdir/vars/*; do
           echo "Reading $varfile"
           source $varfile
 
           echo "Downloading $OCI_SOURCE to $OCI_FILENAME"
-          curl --silent --show-error --location $OCI_SOURCE -o $OCI_FILENAME
+          curl "${CURL_ARGS[@]}" --silent --show-error --location $OCI_SOURCE -o $OCI_FILENAME
```

Comment:
Not directly related to this PR, but I noticed we're not passing `--fail` to curl so we can end up writing an HTTP error response into the output file and proceeding. Although that would be caught by a (somewhat confusing) checksum mismatch

Reply:
Fixed in b0d7434.